pkg/sql/index_selection.go

```diff
@@ -800,6 +800,11 @@ func (v *indexInfo) isCoveringIndex(scan *scanNode) bool {
 	}
 
 	for i, needed := range scan.valNeededForCol {
+		// This is possible during a schema change when we have
+		// additional columns.
+		if i >= len(v.desc.Columns) {
+			return false
```

Comment:
If the column not needed, should we still return false?

Reply:
Hrm, you're right. I'm not too sure why `valNeededForCol` would ever include the mutated columns up to this point (perhaps someone more familiar with schema changes will have more context on this). Is it valid for queries during the planning phase to refer to `TableDescriptor.mutations` columns, but have those columns be resolved during execution? Because if not, maybe we shouldn't initialize `resultColumns, colIdxMap, valNeededForCol` with `n.cols`.